src/ui/party-ui-handler.ts

```diff
@@ -1963,3 +2093,94 @@ class PartyCancelButton extends Phaser.GameObjects.Container {
     this.partyCancelPb.setFrame("party_pb");
   }
 }
+
+class PartyDiscardModeButton extends Phaser.GameObjects.Container {
+  private selected: boolean;
+
+  private transferIcon: Phaser.GameObjects.Sprite;
+  private discardIcon: Phaser.GameObjects.Sprite;
+  private partyDiscardPb: Phaser.GameObjects.Sprite;
+  private textBox: Phaser.GameObjects.Text;
+
+  constructor(x: number, y: number) {
+    super(globalScene, x, y);
+
+    this.setup();
+  }
+
+  setup() {
+    this.transferIcon = globalScene.add.sprite(0, 0, "party_transfer");
+    this.discardIcon = globalScene.add.sprite(0, 0, "party_discard");
+    this.partyDiscardPb = globalScene.add.sprite(-20, 0, "party_pb");
+    this.textBox = addTextObject(-8, -7, i18next.t("partyUiHandler:TRANSFER"), TextStyle.PARTY);
+
+    this.add(this.transferIcon);
+    this.add(this.discardIcon);
+    this.add(this.partyDiscardPb);
+    this.add(this.textBox);
+
+    this.clear();
+  }
+
+  select() {
+    if (this.selected) {
+      return;
+    }
+
+    this.selected = true;
+
+    this.transferIcon.setFrame("selected");
+    this.discardIcon.setFrame("selected");
+    this.partyDiscardPb.setFrame("party_pb_sel");
+  }
+
+  deselect() {
+    if (!this.selected) {
+      return;
+    }
+
+    this.selected = false;
+
+    this.transferIcon.setFrame("normal");
+    this.discardIcon.setFrame("normal");
+    this.partyDiscardPb.setFrame("party_pb");
+  }
+
+  toggleIcon(partyMode: number) {
+    switch (partyMode) {
+      case PartyUiMode.DISCARD:
+        this.transferIcon.setVisible(true);
+        this.discardIcon.setVisible(false);
+        this.partyDiscardPb.setVisible(true);
+        this.textBox.setVisible(true);
+        this.textBox.setText(i18next.t("partyUiHandler:TRANSFER"));
+        this.setPosition(
+          globalScene.currentBattle.double ? 64 : 60,
+          globalScene.currentBattle.double ? -48 : -globalScene.game.canvas.height / 15 - 1,
+        );
+        this.transferIcon.displayWidth = this.textBox.text.length * 9 + 3;
+        break;
+      case PartyUiMode.MODIFIER_TRANSFER:
+        this.transferIcon.setVisible(false);
+        this.discardIcon.setVisible(true);
+        this.partyDiscardPb.setVisible(true);
+        this.textBox.setVisible(true);
+        this.textBox.setText("Discard");
```

Comment:
I'd presume this needs a locale key correct?

```suggestion
        this.textBox.setText("Discard"); // TODO: Add locale key
```